recipe/meta.yaml

```diff
@@ -1,170 +1,254 @@
-    - set MATRIX_GPU_ARCH_TYPE="cuda"                                             # [(pytorch_variant == "gpu") and (win)]
-    - set MATRIX_GPU_ARCH_VERSION="none"                                          # [(pytorch_variant == "cpu") and (win)]
-    - set MATRIX_GPU_ARCH_TYPE="none"                                             # [(pytorch_variant == "cpu") and (win)]
-    - set MATRIX_CHANNEL="defaults"                                               # [win]
-    - set MATRIX_STABLE_VERSION={{ version }}                                     # [win]
-    - set MATRIX_PACKAGE_TYPE="conda"                                             # [win]
-    - set TARGET_OS="windows"                                                     # [win]
-    - python ./smoke_test.py --package torchonly
-    - python ./test/run_test.py --core || true
-    # Run pip check so as to ensure that all pytorch packages are installed
-    # https://github.com/conda-forge/pytorch-cpu-feedstock/issues/24
-    - pip check
-    - python -c "import torch; print(torch.__version__)"
-    # We have had issues with openmp .dylibs being doubly loaded in certain cases. These two tests catch those issues
-    - python -c "import torch; import numpy"
-    - python -c "import numpy; import torch"
-    # distributed support is enabled by default on linux; for mac, we enable it manually in build.sh
-    - python -c "import torch; assert torch.distributed.is_available()"        # [linux or osx]
-    - python -c "import torch; assert torch.backends.mkldnn.m.is_available()"  # [x86 and cuda_compiler_version == "None"]
-    - python -c "import torch; assert torch.backends.cuda.is_built()"          # [(pytorch_variant == "gpu") and (linux and x86_64)]
-    - python -c "import torch; assert torch.backends.cudnn.is_available()"     # [(pytorch_variant == "gpu") and (linux and x86_64)]
-    - python -c "import torch; assert torch.cuda.is_available()"               # [(pytorch_variant == "gpu") and (linux and x86_64)]
-    - python -c "import torch; assert torch.backends.cudnn.enabled"            # [(pytorch_variant == "gpu") and (linux and x86_64)]
-    - python -c "import torch; assert torch.backends.mps.is_built()"           # [(pytorch_variant == "gpu") and (osx and arm64)]
+outputs:
+  # The pytorch-cpu and pytorch-gpu metapackages are packages which the user can use to get the
+  # corresponding pytorch variant. If they install the pytorch package, it will give them the
+  # GPU variant if their platform supports it and the CPU variant otherwise.
+  - name: pytorch-{{ pytorch_variant }}
+    build:
+      noarch: generic
+      skip: True  # [py<38]
+    requirements:
+      run:
+        - pytorch ={{ version }}={{ pytorch_variant }}* # NB pinning exact=True will also pin to the python version
+
+  - name: pytorch
+    script: build_pytorch.sh # [not win]
+    script: bld_pytorch.bat  # [win]
+    build:
+      skip: True  # [py<38]
+      skip: True  # [py!=310]
```

Comment:
what is special about py 310?

Reply:
Or more, what is special with 39 and 311. Because right now it'll only compile on 310.

Reply:
thanks, good catch

Reply:
removed

Reply:
this was there to reduce development time during testing